mne/viz/misc.py

```diff
@@ -827,13 +837,20 @@ def plot_filter(h, sfreq, freq=None, gain=None, title=None, color='#1f77b4',
     from scipy.signal import (
         freqz, group_delay, lfilter, filtfilt, sosfilt, sosfiltfilt)
     import matplotlib.pyplot as plt
+
     sfreq = float(sfreq)
     _check_option('fscale', fscale, ['log', 'linear'])
+    if isinstance(plot, str):
+        plot = [plot]
+    for xi, x in enumerate(plot):
+        _check_option('plot[%d]' % xi, x, ('magnitude', 'delay', 'time'))
```

Comment:
It's starting to seem like this should be a new:
```
_check_iterable_option('plot', plot, ('magintude', 'delay', 'time'))
```
Eventually we can make `_check_iterable_option` smart, like raising an error if it's not iterable for example. There is a risk to over-engineering these sorts of things, but it seems like you would have been able to use it half a dozen times or so, and I imagine there are a lot of other similar cases in our codebase, so seems reasonable to me. WDYT?

If you agree it's good, let's do it in another PR since this is already good to go

Reply:
Yeah sure. I will take a crack it implementing this over the next few days in a different PR.